### PR TITLE
fix: minor fixes in functionality

### PR DESCRIPTION
**Pull request by rkapoor10**
- watch later btn now has add and remove functionality with active state
 - history has a video checker to add a video only once
 - liked video also has a checker
 - fixed watchlater btn on playlist videos

### CODE REVIEW

**Review by fahd23 — APPROVED** [review]
The code looks good and UI is awesome. One Suggestion, remove the commented code if not needed.

File: src/components/cards/VideoCard.jsx [hunk 1]
```diff
@@ -36,18 +38,31 @@ const VideoCard = ({details,setOpenModal, setNewVideo}) => {
...
-          <MdWatchLater />
-        </button>
+
+        {searchVideo(videoState["watchLaterVideos"], details) ? (
+          <button
+            className="child-watchlater-btn on-thumbnail-btns fs-3 active-watchlater"
+            title="add to WatchLater"
+            onClick={handleAddToWatchLater}
+          >
+            {/* <span className="btns-extension fs-1r">WatchLater</span>  */}
```
Comment (tusharkandpal) [inline src/components/cards/VideoCard.jsx:48]: can remove this, if not needed!

File: src/components/cards/VideoCard.jsx [hunk 2]
```diff
@@ -62,11 +77,11 @@ const VideoCard = ({details,setOpenModal, setNewVideo}) => {
...
         />
         <div className="coloumn-flex title-container">
-          <p className="video-main-title">
-            {title}
-          </p>
+          <p className="video-main-title">{title}</p>
           <p className="video-sub-title">{creator}</p>
-          <p className="video-sub-title">{statistics["viewCount"]}M views • {statistics["likeCount"]}K likes</p>
+          <p className="video-sub-title">
+            {statistics["viewCount"]}M views • {statistics["likeCount"]}K likes
```
Comment (tusharkandpal) [inline src/components/cards/VideoCard.jsx:83]: can destruct this `statistics` object above.

File: src/components/cards/VideoCardHorizontal.jsx [hunk 3]
```diff
@@ -9,19 +9,19 @@ import { usePlaylist } from "../../context/PlaylistContext/PlaylistContext";
...
     event.stopPropagation();
-    if(location === "WatchLater"){
-      videoDispatch({type:"REMOVE_FROM_WATCHLATER",payload:details})
-      toast.success("Removed from Watch Later")
-    }
-    else{
-      videoDispatch({type:"ADD_TO_WATCHLATER",payload:details})
-      toast.success("Added to Watch Later")
+    if (searchVideo(videoState["watchLaterVideos"], details)) {
+      videoDispatch({ type: "REMOVE_FROM_WATCHLATER", payload: details });
```
Comment (tusharkandpal) [inline src/components/cards/VideoCardHorizontal.jsx:20]: try to send payload as an `object` of key-value pair.

File: src/pages/VideoPlayerPage.jsx [hunk 4]
```diff
@@ -11,10 +11,12 @@ import { toast } from "react-toastify";
...
 const VideoPlayerPage = () => {
   const { playerId } = useParams();
-  const { videosData, videoDispatch } = useVideo();
+  const { videosData, videoState, videoDispatch } = useVideo();
+  const searchVideo = (arr, video) => arr.includes(video);
   const details = findInVideos(videosData, "channelId", playerId);
+
   useEffect(
-    () => videoDispatch({ type: "ADD_TO_HISTORY", payload: details }),
+    () => !searchVideo(videoState["history"],details) && videoDispatch({ type: "ADD_TO_HISTORY", payload: details }),
```
Comment (tusharkandpal) [inline src/pages/VideoPlayerPage.jsx:19]: can destruct for ease of use. like `{ videoState : { history } }`

**Review by tusharkandpal — APPROVED** [review]
Looks overall clean & well structured !

left few suggestions else good to merge !

File: src/components/cards/VideoCard.jsx (hunk 1 above)
Comment (rkapoor10) [reply src/components/cards/VideoCard.jsx:48]: noted

File: src/components/cards/VideoCard.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/components/cards/VideoCard.jsx:83]: ok

File: src/components/cards/VideoCardHorizontal.jsx (hunk 3 above)
Comment (rkapoor10) [reply src/components/cards/VideoCardHorizontal.jsx:20]: noted will do

File: src/pages/VideoPlayerPage.jsx (hunk 4 above)
Comment (rkapoor10) [reply src/pages/VideoPlayerPage.jsx:19]: ok

**Pull request merged**